### DISCUSSION TITLE
pdf_downloader will not work without installing...

### DISCUSSION THREAD

**Discussion by wisdomlight** [nextcloud/all-in-one#4414] [Questions] [answered]
Hi - the Nextcloud sync client is telling me that the app "pdf_downloader" will not work without installing the following required progrmam: pdftk
it also tells me that 
The app "pdf_downloader" will work better if you install the following suggested helper programs: img2pdf, mhonarc, pandoc, tiff2pdfu, unoconv, wkhtmltopdf, pdf2svg"
Should I just apt install all these programs in the master container?

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-add-os-packages-permanently-to-the-nextcloud-container

**Reply by wisdomlight** [reply]
Hi 
Thank you for putting me on the right track.
Before I make any changes (which might cause things to go bad) please, could you have a look at this please and tell me if the procedure is correct?

---

This is the list of containers and images
```
$ docker ps --format "table {{.Names}}\t{{.Image}}" 
NAMES                           IMAGE
nextcloud-aio-apache            nextcloud/aio-apache:latest
nextcloud-aio-notify-push       nextcloud/aio-notify-push:latest
nextcloud-aio-nextcloud         nextcloud/aio-nextcloud:latest
nextcloud-aio-imaginary         nextcloud/aio-imaginary:latest
nextcloud-aio-clamav            nextcloud/aio-clamav:latest
nextcloud-aio-redis             nextcloud/aio-redis:latest
nextcloud-aio-database          nextcloud/aio-postgresql:latest
nextcloud-aio-collabora         nextcloud/aio-collabora:latest
nextcloud-aio-mastercontainer   nextcloud/all-in-one:latest
```
1. `docker stop nextcloud-aio-mastercontainer`
2. `docker rm nextcloud-aio-mastercontainer`
3. `docker rmi nextcloud/all-in-one:latest`

DATADIR and Backup Dir are on external drive:
According to my current AIO interface page 
>Nextcloud's datadir is getting stored in the /media/sherab/ncdata/nextcloud directory. See the NEXTCLOUD_DATADIR documentation on how to change this.

**Q.** Do I need to add this line `--env NEXTCLOUD_DATADIR="/media/sherab/ncdata/nextcloud"` to the reinstalling of the all-in-one:latest?

> Backups get created in the following directory on the host: /media/sherab/ncbackup/nextcloudbkp/borg

**Q.** How do I inform the docker run about the location of my backup folder?

And to permanently add the needed packages `--env NEXTCLOUD_ADDITIONAL_APKS="imagick extension1 extension2"`

So is this the correct command to run in order to reinstall and permamently add new packages as well as ensuring nextcloud will find the Backup and DATADIR on the external HD?

```
sudo docker run \
--init \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 80:80 \
--publish 8080:8080 \
--publish 8443:8443 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
--env NEXTCLOUD_DATADIR="/media/sherab/ncdata/nextcloud"\
--env NEXTCLOUD_ADDITIONAL_APKS="img2pdf  mhonarc  pandoc  tiff2pdfu  unoconv  wkhtmltopdf  pdf2svg" \
nextcloud/all-in-one:latest
```
